Backoffice/Security/Authorization/Voter/AbstractNodeVoter.php

```diff
@@ -25,45 +25,50 @@
      * Vote for Read action
      * A user can read a node if it is in his perimeter
      *
-     * @param NodeInterface $node
-     * @param UserInterface $user
+     * @param NodeInterface  $node
+     * @param TokenInterface $token
      *
      * @return bool
      */
-    protected function voteForReadAction($node, UserInterface $user)
+    protected function voteForReadAction($node, TokenInterface $token)
     {
-        return $user->hasRole(ContributionRoleInterface::NODE_CONTRIBUTOR)
+        $user = $token->getUser();
+
+        return $this->hasRole($token, ContributionRoleInterface::NODE_CONTRIBUTOR)
             && $this->isSubjectInPerimeter($this->getPath($node), $user, NodeInterface::ENTITY_TYPE);
     }
 
     /**
      * Vote for $action on $node owned by $user
      * A user can act on his own nodes if he has the NODE_CONTRIBUTOR role and the node is in his perimeter 
      *
-     * @param string        $action
-     * @param NodeInterface $node
-     * @param UserInterface $user
+     * @param string         $action
+     * @param NodeInterface  $node
+     * @param TokenInterface $token
      *
      * @return bool
      */
-    protected function voteForOwnedSubject($action, $node, UserInterface $user)
+    protected function voteForOwnedSubject($action, $node, TokenInterface $token)
     {
-        return $user->hasRole(ContributionRoleInterface::NODE_CONTRIBUTOR)
+        $user = $token->getUser();
+
+        return $this->hasRole($token, ContributionRoleInterface::NODE_CONTRIBUTOR)
             && $this->isSubjectInPerimeter($this->getPath($node), $user, NodeInterface::ENTITY_TYPE);
     }
 
     /**
      * Vote for $action on $node not owned by $user
      * A user can act on someone else's node if he has the matching super role and the node is in his perimeter
      *
-     * @param string        $action
-     * @param NodeInterface $node
-     * @param UserInterface $user
+     * @param string         $action
+     * @param NodeInterface  $node
+     * @param TokenInterface $token
      *
      * @return bool
      */
-    protected function voteForSomeoneElseSubject($action, $node, UserInterface $user)
+    protected function voteForSomeoneElseSubject($action, $node, TokenInterface $token)
     {
+        $user = $token->getUser();
```

Comment:
@alavieille : there's no more the need to create the local var $user, you can directly inject $token->getUser() in the condition, or better, you can inject the $token in `isSubjectInPerimeter`. That way if we refacto the token to store the perimeter, the modification here will be small